lib/modules.nix

```diff
@@ -293,16 +282,19 @@ rec {
              current option declaration as the file use for the submodule.  If the
              submodule defines any filename, then we ignore the enclosing option file. */
           options' = toList opt.options.options;
+
+          packSubmodule = file: m:
```

Comment:
maybe inline? I get little speedup after inlining this.

Reply:
I don't want to inline because it would duplicate this three times. I was however able to push these definitions outside of `mergeOptionDecls` since they don't depend on any arguments, saving some thunks and memory. See https://github.com/NixOS/nixpkgs/pull/75031/commits/093768cc5464c654ecaa19fb19a8d6ad4b431a70